lib/eventum/auth/APIAuthToken.php

```diff
@@ -0,0 +1,159 @@
+// | You should have received a copy of the GNU General Public License    |
+// | along with this program; if not, write to:                           |
+// |                                                                      |
+// | Free Software Foundation, Inc.                                       |
+// | 51 Franklin Street, Suite 330                                        |
+// | Boston, MA 02110-1301, USA.                                          |
+// +----------------------------------------------------------------------+
+
+class APIAuthToken
+{
+    private static $default_alg = 'HS256';
+
+    public static function generate($usr_id)
+    {
+        $factory = new RandomLib\Factory();
+        $generator = $factory->getMediumStrengthGenerator();
+        $token = $generator->generateString(32, \RandomLib\Generator::CHAR_ALNUM);
+
+        self::saveToken($usr_id, $token);
+
+        return $token;
+    }
+
+    public static function saveToken($usr_id, $token)
+    {
+        $sql = 'INSERT INTO
+                    {{%api_token}}
+                SET
+                    apt_usr_id = ?,
+                    apt_created = ?,
+                    apt_token = ?';
+        try {
+            $res = DB_Helper::getInstance()->query($sql, array(
+                $usr_id,
+                Date_Helper::getCurrentDateGMT(),
+                $token,
+            ));
+        } catch (DbException $e) {
+            return -1;
+        }
+    }
+
+    public static function isTokenValidForEmail($token, $email)
+    {
+        try {
+            if (self::checkTokenStatus($token) == User::getUserIDByEmail($email, true)) {
+                return true;
+            }
+        } catch (AuthException $e) {
+            return false;
+        }
+    }
+
+    public static function checkTokenStatus($token)
+    {
+        $sql = "SELECT
+                    apt_usr_id
+                FROM
+                    {{%api_token}}
+                WHERE
+                    apt_token = ? AND
+                    apt_status = 'active'";
+        try {
+            $usr_id = DB_Helper::getInstance()->getOne($sql, array($token));
+        } catch (DbException $e) {
+            throw new AuthException('Error fetching user token');
+        }
+        if (empty($usr_id)) {
+            throw new AuthException('Invalid token');
+        }
+
+        return $usr_id;
+    }
+
+    public static function getUserIDByToken($token)
+    {
+        try {
+            // check that token is valid and not expired
+            $decoded = JWT::decode($token, Auth::privateKey(), array(self::$default_alg));
+
+            // make sure token hasn't been revoked
+            $usr_id = self::checkTokenStatus($token);
+
+            return $usr_id;
+        } catch (Exception $e) {
+            throw new AuthException($e);
+        }
+    }
+
+    public static function getTokensForUser($usr_id, $auto_generate = false)
```

Comment:
I think this is the best place for the auto generate logic. Otherwise it is moved out into the controller and if we need to auto generate a token in some other place we will have to duplicate it.

No code would use the $active parameter at this point and I prefer not to add dead code.